class/practice5.1/MegaData.h

```diff
@@ -0,0 +1,30 @@
+#ifndef MEGADATA_H
+#define MEGADATA_H
+
+#include <vector>
+#include <memory>
+
+class MegaData {
+public:
+    std::vector<float> smallArray;
+    std::vector<double> bigArray;
+
+    MegaData();
+    void reset();
+};
+
+class MegaDataPool {
+private:
+    std::vector<std::unique_ptr<MegaData>> pool;
+    std::vector<MegaData*> available;
+
+public:
+    explicit MegaDataPool(size_t size);
+    MegaData* acquire();
```

Comment:
use [[nodiscard]] for this function - we don't want someone to forget to use the result. If someone will accidentally call `acquire();` without assignment then the returned pointer will not be put to release function and we will not be able to use it.